### PR TITLE
Board handler pull request

### PR DESCRIPTION
**Pull request by xKoem**
Created BoardHandler microservice with tests.

### CODE REVIEW

**Comment by boczkas** [comment]
Build 140 Successful

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerShootCoordinateEventListener.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    public void onApplicationEvent(PlayerShootCoordinateEvent event) {
+
+        PlayerShootCoordinate playerShootCoordinate = event.getCoordinate();
+
+        Player player = playerShootCoordinate.getPlayer();
+        Coordinate shotCoordinate = playerShootCoordinate.getCoordinate();
+        FieldState opponentFieldState = null;
+
+        //TODO: creation of this ugly construction was caused by the fact that
+        //TODO: getOpponentFieldStatus and markHit methods can throw an exception
```
Comment (krzysieksr) [inline BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerShootCoordinateEventListener.java]: Please refactor this in the future

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/Opponents.java [hunk 2]
```diff
@@ -2,21 +2,35 @@
...
 
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 
 class Opponents {
     private List<Player> opponents;
 
+    Opponents(Player player) {
+        opponents = new ArrayList<>();
```
Comment (krzysieksr) [inline BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/Opponents.java]: initiallize ArralyList two lines higher

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerConfigurationEventListener.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    public PlayerConfigurationEventListener(BoardHandler boardHandler) {
+        this.boardHandler = boardHandler;
+    }
+
+    @Override
+    public void onApplicationEvent(PlayerConfigurationEvent event) {
+        PlayerConfiguration playerConfiguration = event.getPlayerConfiguration();
+        Player player = playerConfiguration.getPlayer();
+        boardHandler.addPlayer(player);
+        boardHandler.addPlayerToRoom(new Room("1"), player);
```
Comment (totine) [inline BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerConfigurationEventListener.java]: magic string :)

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/AllRoomsOpponents.java [hunk 4]
```diff
@@ -1,21 +1,20 @@
...
 
     public AllRoomsOpponents() {
-        this.allRoomsOpponents = new ArrayList<>();
+        this.allRoomsOpponents = new HashMap<>();
     }
 
     public void addOpponents(Player player, Player opponent) {
         Opponents opponents = new Opponents(player, opponent);
-        allRoomsOpponents.add(opponents);
+        allRoomsOpponents.put(new Room("1"), opponents);
```
Comment (totine) [inline BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/AllRoomsOpponents.java]: magic string :)

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/AllRoomsOpponents.java [hunk 5]
```diff
@@ -27,15 +26,29 @@ public String toString() {
...
+        if (foundOpponents.isPresent()) {
             Opponents opponents = foundOpponents.get();
             return opponents.getOpponent(player);
         }
 
         throw new NoSuchPlayerException(player);
     }
+
+    public void addPlayerToRoom(Room room, Player player) {
+        if(!allRoomsOpponents.containsKey(room)){
```
Comment (totine) [inline BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/AllRoomsOpponents.java]: space

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/Opponents.java [hunk 6]
```diff
@@ -2,21 +2,35 @@
...
+        opponents.add(player);
+    }
+
     Opponents(Player player, Player opponent) {
         opponents = Arrays.asList(player, opponent);
     }
 
+    public void addOpponent(Player player) {
```
Comment (totine) [inline BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/Opponents.java]: if there is 2 players in list then maybe some exception throwing ?

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/PlayersBoards.java [hunk 7]
```diff
@@ -53,6 +54,28 @@ public Board getBoard(Player player) {
...
+        PlayerBoards playerBoards = createPlayerBoards(fleet);
+        playersBoards.put(player, playerBoards);
+    }
+
+    public void addFleetForPlayer(Player player, Fleet fleet) {
+        PlayerBoards playerBoards = createPlayerBoards(fleet);
+        playersBoards.put(player, playerBoards);
+    }
+
+    int getMapSize() {
```
Comment (totine) [inline BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/PlayersBoards.java]: getSize

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/service/BoardHandler.java [hunk 8]
```diff
@@ -19,22 +20,22 @@
     private final PlayersBoards playersBoards;
     private final AllRoomsOpponents allRoomsOpponents;
 
-    public BoardHandler() {
-        this.allRoomsOpponents = new AllRoomsOpponents();
-        this.playersBoards = new PlayersBoards();
+    public BoardHandler(AllRoomsOpponents roomOpponents, PlayersBoards playersBoards) {
```
Comment (totine) [inline BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/service/BoardHandler.java]: :+1:

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerShootCoordinateEventListener.java (hunk 1 above)
Comment (boczkas) [reply BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerShootCoordinateEventListener.java]: Will do

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/Opponents.java (hunk 2 above)
Comment (boczkas) [reply BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/Opponents.java]: done

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerConfigurationEventListener.java (hunk 3 above)
Comment (boczkas) [reply BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerConfigurationEventListener.java]: fixed

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/AllRoomsOpponents.java (hunk 4 above)
Comment (boczkas) [reply BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/AllRoomsOpponents.java]: refactored

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/AllRoomsOpponents.java (hunk 5 above)
Comment (boczkas) [reply BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/AllRoomsOpponents.java]: fixed

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/Opponents.java (hunk 6 above)
Comment (boczkas) [reply BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/Opponents.java]: Will be done later.
Tracked under #140

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/PlayersBoards.java (hunk 7 above)
Comment (boczkas) [reply BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/PlayersBoards.java]: done

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/service/BoardHandler.java (hunk 8 above)
Comment (boczkas) [reply BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/service/BoardHandler.java]: danke

**Pull request merged**